### PR TITLE
Fix matrix multiplication logic for transforms

### LINKED ISSUE #10111: Transforms no longer compose correctly
### Problem Description

Previously, we composed transforms in JS using processTransform.js from the upstream RN repo. As of #9798, we forked the processTransform.js to allow the RNW Fabric implementation to compose transforms natively. Unfortunately, the float4x4 matrix multiply function does not produce consistent results with iOS and Android (see screenshots below).

### Steps To Reproduce

1. Run the RNTester > APIs > Transforms example
2. Compare rendering between Windows and iOS/Android
3. Notice that the two platforms do not produce identical results when transforms (e.g., translate and rotate) are composed.

### Expected Results

Transforms must compose identically on both platforms.

### CLI version

npx react-native --version

### Environment

```markdown
npx react-native info
```

### Target Platform Version

_No response_

### Target Device(s)

_No response_

### Visual Studio Version

_No response_

### Build Configuration

_No response_

### Snack, code example, screenshot, or link to a repository
|iOS|Windows|
|----|----|
|Screen Shot 2022-06-14 at 1 04 06 PM|image
|
|Screen Shot 2022-06-14 at 1 06 04 PM|image|

### PR DESCRIPTION
**Pull request by rozele**
Fixes #10111

### Type of Change
_Erase all that don't apply._
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

### Why
As reported in #10111, float4x4's operator* does not produce the same output as expected on other RN platforms. 

Resolves #10111

### What
This change ports the logic from RN's MatrixMath.js to ensure multiplication of transforms produces the same result on Windows as on iOS and Android.

## Screenshots
|Before|After|iOS|
|----|----|----|
|image|image|Screen Shot 2022-06-14 at 1 04 06 PM|
|image|image|Screen Shot 2022-06-14 at 1 06 04 PM|

One more for good measure - notice that the JS driven animation just sends transform props (rather than computed matrix) now that we forked `processTransform.windows.js`, so the animation doesn't match expectations from the native driver / iOS:

*Before:*

https://user-images.githubusercontent.com/1106239/173640312-3a7531ff-b9a7-48b3-9b96-e24b50882c44.mp4

*After:*

https://user-images.githubusercontent.com/1106239/173640470-0368c679-c365-4d00-99d6-b2839d80c242.mp4

*iOS:*

https://user-images.githubusercontent.com/1106239/173641318-2be0f961-f12b-44e5-b0da-f8afeced282b.mov

## Testing
See screenshots / recordings above, confirmed that transforms compose as expected on iOS.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/10112)

### CODE REVIEW

**Review by acoates-ms — APPROVED** [review]

**Review by asklar — CHANGES_REQUESTED** [review]
marking requesting changes so we can understand where the difference is coming from

File: vnext/Microsoft.ReactNative/Views/FrameworkElementViewManager.cpp [hunk 1]
```diff
@@ -181,52 +226,60 @@ bool FrameworkElementViewManager::UpdateProperty(
                 innerMatrix.m42 = static_cast<float>(innerValue[13].AsDouble());
                 innerMatrix.m43 = static_cast<float>(innerValue[14].AsDouble());
                 innerMatrix.m44 = static_cast<float>(innerValue[15].AsDouble());
-                transformMatrix = transformMatrix * innerMatrix;
```
[the remark this answers is not in the document]
Comment (asklar) [reply vnext/Microsoft.ReactNative/Views/FrameworkElementViewManager.cpp:184]: IOW in matrices, A*B != B*A, so maybe we are applying the product in the wrong order

[the remark this answers is not in the document]
Comment (rozele) [reply vnext/Microsoft.ReactNative/Views/FrameworkElementViewManager.cpp:184]: Good point - I thought matrix multiplication was commutative, but it is indeed not. I'll check.

**Comment by rozele** [comment]
> marking requesting changes so we can understand where the difference is coming from

Thanks @asklar. It was indeed an issue with ordering, iOS and Android multiply in the reverse order. Fixed, but left the `MultiplyInto` helper so hopefully future transform types do not mess up the argument order for multiplication.

**Review by asklar — APPROVED** [review]

**Comment by acoates-ms** [comment]
@rozele , any chance you can port this to the 0.69 branch too?

**Pull request merged**